### ISSUE TITLE
Exception when setting properties in Grails 3.3.0.M2 

### ISSUE THREAD

**Issue by erichelgeson** [longwa/build-test-data#76]
It looks like the API may have changed for domain classes in 3.3.0 - 

When setting a property such as:

```groovy
User.build(email:"foo@bar.com")
```
I get: 
```
Cannot get property 'email' on null object
java.lang.NullPointerException: Cannot get property 'email' on null object
	at grails.buildtestdata.DomainInstanceBuilder.setDomainPropertyValue(DomainInstanceBuilder.groovy:179)
	at grails.buildtestdata.DomainInstanceBuilder.populateInstance(DomainInstanceBuilder.groovy:148)
	at grails.buildtestdata.DomainInstanceBuilder.build(DomainInstanceBuilder.groovy:136)
	at grails.buildtestdata.DomainInstanceBuilder.build(DomainInstanceBuilder.groovy:135)
	at grails.buildtestdata.BuildTestDataService.addBuildMethods_closure5(BuildTestDataService.groovy:45)
	at groovy.lang.Closure.call(Closure.java:414)
	at com.myapp.RegisterControllerForgotPasswordSpec.forgotPassword POST - valid user(RegisterControllerForgotPasswordSpec.groovy:46)
```

Unsure if the fix should be on b-t-d or grails - cc @graemerocher

related: https://github.com/longwa/build-test-data/issues/74

**Comment by erichelgeson** [comment]
Are there any plans to maintain this for the Grails 3.3.x line? Writing tests without it is a bit of a pain :)

**Comment by drennane** [comment]
Bump! Experiencing the same issue. I also use it for seeding. Pity to lose it.

**Comment by longwa** [comment]
I'm working on a 3.3.x version of BTD at the moment. Our company is just now moving to this release so I haven't had any chance to really address this.

It will be a breaking change, switching to a Trait based system instead of annotations and metaClass methods.

**Comment by erichelgeson** [comment]
@longwa - @tkvw had a similar idea, though not sure how far he got/is going: https://github.com/tkvw/gorm-build-test-data

Great to hear you're working on 3.3.x line!

**Comment by longwa** [comment]
That's funny. We started about the same time and have some similarities in
our implementation as well. Thanks for the heads up.

**Comment by longwa** [comment]
Version 3.3.0.RC1 has been released for Grails 3.3.0 and higher. Still planning on bringing in some of what @tkvw did in his gorm port.

**Comment by erichelgeson** [comment]
Just updated my project and went well! Only one minor issue I'll open in a min.

Thanks for all the working on making Grails testing Gr8 :)
